Sources/NIOSOCKS/Messages/ClientRequest.swift

```diff
@@ -81,28 +76,25 @@ public struct Command: Hashable {
 // MARK: - AddressType
 
 /// The address used to connect to the target host.
-public enum AddressType: Hashable {
-    
-    /// An IPv4 address (4 bytes), e.g. *192.168.1.2*
-    case ipv4([UInt8])
-    
-    /// A fullly-qualified domain name, e.g. *apple.com*
-    case domain([UInt8])
+public struct AddressType: Hashable {
```

Comment:
Do we need to expose this type publicly? It seems to add the size property, but apart from this? Looks to me like it could also be an internal extension on `SocketAddress`?

Reply:
This type is part of the API of the handler.

Reply:
Yes, this is true, however I can't see a reason, why we don't use `SocketAddress` in the handler directly?

Reply:
Because the handler can take domain names too. SOCKS explicitly allows you to ask to connect to a domain without resolving it yourself.

Reply:
Yes Cory is right - let's discuss this in standup briefly